apps/ui-library/content/docs/infinite-query-hook.mdx

```diff
@@ -0,0 +1,271 @@
+---
+title: Infinite Query Hook
+description: React hook for infinite lists, fetching data from Supabase.
+---
+
+<BlockPreview name="infinite-list-demo" />
+
+## Installation
+
+<BlockItem
+  name="infinite-query-hook"
+  description="Installs the Infinite List component and necessary Supabase client setup."
+/>
+
+## Folder structure
+
+<RegistryBlock itemName="infinite-query-hook" />
+
+## Introduction
+
+The Infinite Query Hook provides a single React hook which will make it easier to load data progressively from your Supabase database. It handles data fetching and pagination state, It is meant to be used with infinite lists or tables.
+The hook is fully typed, provided you have generated and setup your database types.
+
+## Adding types
+
+Before using this hook, we **highly** recommend you setup database types in your project. This will make the hook fully-typesafe. More info about generating Typescript types from database schema [here](https://supabase.com/docs/guides/api/rest/generating-types)
```

Comment:
I think worth also adding:

Add the definitions to your Supabase client(s) as described [here](https://supabase.com/docs/guides/api/rest/generating-types#using-typescript-type-definitions).